resolwe/flow/models/data.py

```diff
@@ -259,7 +259,8 @@ def save(self, render_name=False, *args, **kwargs):
 
         self.save_storage(self.output, self.process.output_schema)  # pylint: disable=no-member
 
-        hydrate_size(self)
+        if self.status != Data.STATUS_ERROR:
```

Comment:
At the moment only output validation and file size hydration are skipped. If any other step will fail, also error handling will fail. One example is that invalid json is stored to json field. But do we want to create storage objects also for failed object? Or at least try and leave it it data object if it fails?

Anyway this will be quite challenging to render because of inconsistencies.